near-contract-standards/src/non_fungible_token/mod.rs

```diff
@@ -1,15 +1,36 @@
+/// Trait for the [approval management standard](https://nomicon.io/Standards/NonFungibleToken/ApprovalManagement.html) for NFTs.
 pub mod approval;
+/// Common implementation of the [approval management standard](https://nomicon.io/Standards/NonFungibleToken/ApprovalManagement.html) for NFTs.
 pub mod approval_impl;
+/// Approval receiver is the trait for the method called (or attempted to be called)
+/// on the contract/account that has just been approved. This is not required to implement.
 pub mod approval_receiver;
+/// Trait for the [core non-fungible token standard](https://nomicon.io/Standards/NonFungibleToken/Core.html).
 pub mod core;
+/// Common implementation of the [core non-fungible token standard](https://nomicon.io/Standards/NonFungibleToken/Core.html).
 pub mod core_impl;
+/// Trait for the [NFT enumeration standard](https://nomicon.io/Standards/NonFungibleToken/Enumeration.html).
+/// This provides useful view-only methods returning token supply, tokens by owner, etc.
 pub mod enumeration;
+/// Common implementation of the [NFT enumeration standard](https://nomicon.io/Standards/NonFungibleToken/Enumeration.html).
 pub mod enumeration_impl;
+/// Macros typically used by a contract wanting to take advantage of the non-fungible
+/// token NEAR contract standard approach.
 pub mod macros;
+/// Metadata traits and implementation according to the [NFT enumeration standard](https://nomicon.io/Standards/NonFungibleToken/Metadata.html).
+/// This covers both the contract metadata and the individual token metadata.
 pub mod metadata;
+/// Trait for the receiver of an `nft_transfer_call` method call. This method is
+/// implemented on a receiving account separate from the NFT contract. This is
+/// part of the core standard.
 pub mod receiver;
+/// Trait for the resolver of an `nft_transfer_call` method call. This method is
+/// implemented on the NFT contract and used to resolve how the transfer-and-call
+/// process went. This is part of the core standard.
 pub mod resolver;
+/// The Token struct for the non-fungible token.
 pub mod token;
+/// NFT utility functions
```

Comment:
if only the trait or specific types are needed by a developer, why expose all modules themselves? would just cause a more annoying import process if it doesn't autocomplete imo.

This just means it will be harder to maintain without breaking change because you can never move anything out of those modules or delete any of them. Might make sense for things like receiver, resolver, token but it's hard for me to justify exporting things like `core`, `core_impl`, `enumeration_impl`

Reply:
also just read https://github.com/near/near-sdk-rs/pull/361#discussion_r640283800 comment. While I think it can definitely be done in another pass, we have to make sure that pass is done before a release or else there would be API breaking changes for this sort of re-export

Reply:
Cool thanks, can we sync on this? I believe I could use a little guidance on how it might be best laid out. Super happy to shift this around and thanks for pointing out the breaking change aspect.

Reply:
Resolving conversation as the changes have been moved to https://github.com/near/near-sdk-rs/pull/435